Nodejs/Product/Nodejs/Options/NodejsNpmOptionsControl.cs

```diff
@@ -42,20 +36,26 @@ public partial class NodejsNpmOptionsControl : UserControl {
         }
 
         private void ClearCacheButton_Click(object sender, EventArgs e) {
+            bool didClearNpmCache = DeleteCacheDirectory("npm cache", NodejsConstants.NpmCachePath);
+            bool didClearTools = DeleteCacheDirectory("NTVS external tools", NodejsConstants.ExternalToolsPath);
+
+            _cacheClearedSuccessfully.Visible = (didClearNpmCache && didClearTools);
+        }
+
+        private static bool DeleteCacheDirectory(string displayName, string cachePath) {
             try {
-                Directory.Delete(_npmCachePath, true);                    
-                _cacheClearedSuccessfully.Visible = true;
+                Directory.Delete(cachePath, true);
+               return true;
             } catch (DirectoryNotFoundException) {
                 // Directory has already been deleted. Do nothing.
-                _cacheClearedSuccessfully.Visible = true;
+                return true;
             } catch (IOException exception) {
                 // files are in use or path is too long
```

Comment:
Huh, just now realizing this case might be pretty likely in the case of `typings`... what's the max path length of typings?